### PR TITLE
Extract messages to separate file #GoodnessSquad

### PR DESCRIPTION
**Pull request by reisel**
First stage of issue: #653 
I've extracted messages to a separate file in order to substitute them in the production build.
extracted 5.8kb of data.

### CODE REVIEW

File: src/core/derivation.ts [hunk 1]
```diff
@@ -103,8 +105,8 @@ export function isComputingDerivation() {
 export function checkIfStateModificationsAreAllowed() {
 	if (!globalState.allowStateChanges) {
 		invariant(false, globalState.strictMode
-			? "It is not allowed to create or change state outside an `action` when MobX is in strict mode. Wrap the current method in `action` if this state change is intended"
-			: "It is not allowed to change the state when a computed value or transformer is being evaluated. Use 'autorun' to create reactive functions with side-effects."
+			? getMessage("m030")
+			:getMessage("m031")
```
Comment (A-gambit) [inline src/core/derivation.ts]: add space after `:`

File: src/utils/messages.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+"m038" : "Missing items in this list?\n	  1. Check whether all used values are properly marked as observable (use isObservable to verify)\n	  2. Make sure you didn't dereference values too early. MobX observes props, not primitives. E.g: use 'person.name' instead of 'name' in your computation.",
+
+}
+
+export function getMessage(id : string) {
+	return messages[id];
+};
```
Comment (A-gambit) [inline src/utils/messages.ts]: Add new line

File: src/utils/messages.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+"m038" : "Missing items in this list?\n	  1. Check whether all used values are properly marked as observable (use isObservable to verify)\n	  2. Make sure you didn't dereference values too early. MobX observes props, not primitives. E.g: use 'person.name' instead of 'name' in your computation.",
+
```
Comment (A-gambit) [inline src/utils/messages.ts]: Remove empty line

Comment (A-gambit) [inline src/utils/messages.ts]: Remove empty line

File: src/utils/messages.ts (hunk 2 above)
Comment (reisel) [inline src/utils/messages.ts]: where?

**Comment by mweststrate** [comment]
Merged and will be part of next release! Probably needs some follow up to leverage this new mechanism during production builds, (`NODE_ENV === production` like react?), and can be expanded for parameterizable actions

**Pull request merged**